### PR TITLE
feat(multichain): enhance search api

### PR DESCRIPTION
**Pull request by lok52**
* Integrated Dapp search into quick search. In this case aggregator just proxies search requests directly to Dapp Marketplace service and then formats the response.
* Added string prefix search for addresses. This one also supports pagination via `/api/v1/addresses`. In quick search it yields only top 100 matches by default.

## Release Notes

- **New Features**
	- Introduced a new endpoint to list addresses via the API.
	- Added support for decentralized applications (DApps) with a new client and associated data structures.
	- Enhanced search functionality to include DApps in results.
	- Implemented pagination for address queries.
	- Added a new field for marketplace DApps in search results.
	- Updated configuration to specify DApp client settings.

- **Bug Fixes**
	- Improved error handling in search queries.

- **Documentation**
	- Updated API specifications to reflect new features and endpoints.
	- Revised environment variable documentation for clarity.

- **Chores**
	- Added new dependencies to support enhanced functionalities.

### CODE REVIEW

File: multichain-aggregator/multichain-aggregator-server/src/server.rs [hunk 1]
```diff
@@ -65,11 +65,21 @@ pub async fn run(settings: Settings) -> Result<(), anyhow::Error> {
         .fetch_all()
         .await?
         .into_iter()
-        .map(|(id, chain)| (id as i64, chain).into())
+        .filter_map(|(id, chain)| {
+            let id = id.parse::<i64>().ok()?;
```
Comment (bragov4ik) [inline multichain-aggregator/multichain-aggregator-server/src/server.rs:69]: the decision was to ignore the chain ids with prefixes for now?

File: multichain-aggregator/multichain-aggregator-server/src/services/multichain_aggregator.rs [hunk 2]
```diff
@@ -1,31 +1,47 @@
...
-use multichain_aggregator_proto::blockscout::multichain_aggregator::v1::{
-    QuickSearchRequest, QuickSearchResponse,
+    self as logic, api_key_manager::ApiKeyManager, dapp_client::DappClient, error::ServiceError,
+    Chain,
 };
 use sea_orm::DatabaseConnection;
+use std::str::FromStr;
 use tonic::{Request, Response, Status};
 
+const DEFAULT_PAGE_SIZE: u32 = 50;
```
Comment (bragov4ik) [inline multichain-aggregator/multichain-aggregator-server/src/services/multichain_aggregator.rs:14]: what do you think about making it a setting as well?

File: multichain-aggregator/multichain-aggregator-server/src/services/multichain_aggregator.rs [hunk 3]
```diff
@@ -58,19 +74,75 @@ impl MultichainAggregatorService for MultichainAggregator {
...
+        &self,
+        request: Request<ListAddressesRequest>,
+    ) -> Result<Response<ListAddressesResponse>, Status> {
+        let inner = request.into_inner();
+
+        let page_token: Option<(alloy_primitives::Address, logic::ChainId)> =
+            inner.page_token.map(parse_query_2).transpose()?;
+        let page_size = self.normalize_page_size(inner.page_size);
+
+        let (addresses, next_page_token) = logic::repository::addresses::search_by_query_paginated(
```
Comment (bragov4ik) [inline multichain-aggregator/multichain-aggregator-server/src/services/multichain_aggregator.rs:87]: Nit: it confuses me a bit that 'search' is called within 'list'. So, the endpoint seems more like a search to me 🤔

**Review by bragov4ik — APPROVED** [review]
LGTM, very nice and understandable 👍

File: multichain-aggregator/multichain-aggregator-server/src/server.rs (hunk 1 above)
Comment (lok52) [reply multichain-aggregator/multichain-aggregator-server/src/server.rs:69]: Yes, we want to skip chains with non-integer chain ids

**Pull request merged**